### PR TITLE
Add AnthropicStreamingChatModel

### PR DESCRIPTION
**Pull request by ilopezluna**
Follow up of [integration with Anthropic](https://github.com/langchain4j/langchain4j/pull/727) to include streaming chat model.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method for handling streaming messages in the Anthropic API.
	- Added functionality for streaming messages and improved error handling in the Anthropic client.
	- New classes `Message` and `Delta` added for better message response handling.
	- Implemented a streaming chat language model for generating AI responses based on user messages.
- **Tests**
	- Added tests for the new streaming chat language model, covering functionality such as streaming answers and handling different types of system messages.

### CODE REVIEW

**Comment by langchain4j** [comment]
Hi @ilopezluna, thanks a lot for your contribution! I and @kuraleta have been working lately on the same thing and it is a pity that  the work that was done is duplicated. [This issue](https://github.com/langchain4j/langchain4j/issues/714) was assigned to @kuraleta, so please let's sync next time before implementation? Thanks a lot for understanding!

**Comment by langchain4j** [comment]
@ilopezluna could you please review https://github.com/langchain4j/langchain4j/pull/750 ? 🙏

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicClient.java [hunk 1]
```diff
@@ -78,7 +89,39 @@ AnthropicCreateMessageResponse createMessage(AnthropicCreateMessageRequest reque
...
+        Call<ResponseBody> call = anthropicApi.streamingMessages(apiKey, version, request);
+        try (InputStream bodyStream = call.execute().body().byteStream()) {
+            BufferedReader input = new BufferedReader(new InputStreamReader(bodyStream));
+            String line;
+            while ((line = input.readLine()) != null) {
+                if (line.startsWith("data:")) {
+                    String json = line.substring(5);
+                    AnthropicCreateMessageResponse response = GSON.fromJson(json, AnthropicCreateMessageResponse.class);
+                    if ("message_start".equals(response.getType())) {
+                        inputTokens = response.getMessage().getUsage().getInputTokens();
```
Comment (langchain4j) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicClient.java:105]: `message_start` contains 1 output token, at least in my tests:

`DEBUG: onEvent() type: 'message_start', data: {"type":"message_start","message":{"id":"msg_01P9YFhdBDD8Rkkfpvvr5oe9","type":"message","role":"assistant","content":[],"model":"claude-3-sonnet-20240229","stop_reason":null,"stop_sequence":null,"usage":{"input_tokens":8,"output_tokens":1}}}`

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicClient.java [hunk 2]
```diff
@@ -78,7 +89,39 @@ AnthropicCreateMessageResponse createMessage(AnthropicCreateMessageRequest reque
...
+                        String next = response.getDelta().getText();
+                        sb.append(next);
+                        handler.onNext(next);
+                    }
+                    else if ("error".equals(response.getType())) {
+                        handler.onError(new RuntimeException(response.getDelta().getText()));
+                    } else if ("message_delta".equals(response.getType()) && response.getUsage() != null) {
+                        TokenUsage tokenUsage = new TokenUsage(inputTokens, response.getUsage().getOutputTokens());
+                        AiMessage aiMessage = AiMessage.aiMessage(sb.toString());
+                        handler.onComplete(new dev.langchain4j.model.output.Response<>(aiMessage, tokenUsage, FinishReason.STOP));
```
Comment (langchain4j) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicClient.java:117]: an actual finish reason is present in `message_delta` event: 

`DEBUG: onEvent() type: 'message_delta', data: {"type":"message_delta","delta":{"stop_reason":"max_tokens","stop_sequence":null},"usage":{"output_tokens":1}}`

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicClient.java (hunk 1 above)
Comment (ilopezluna) [reply langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicClient.java:105]: You're right, also [in their docs](https://docs.anthropic.com/claude/reference/messages-streaming)
```
message_start
data: {"type": "message_start", "message": {"id": "msg_1nZdL29xx5MUA1yADyHTEsnR8uuvGzszyY", "type": "message", "role": "assistant", "content": [], "model": "claude-3-opus-20240229, "stop_reason": null, "stop_sequence": null, "usage": {"input_tokens": 25, "output_tokens": 1}}}

```

**Pull request closed without merging**